docs/rules/max-statements.md

```diff
@@ -54,16 +90,47 @@ function foo() {
     return 42;
   };
 }
-```
 
-```js
-/*eslint max-statements: ["error", 1, {ignoreTopLevelFunctions: true}]*/  // Maximum of 1 statement.
-(function() {
-  var bar = 1;
+let foo = () => {
+  var foo1 = 1;
+  var foo2 = 2;
+  var foo3 = 3;
+  var foo4 = 4;
+  var foo5 = 5;
+  var foo6 = 6;
+  var foo7 = 7;
+  var foo8 = 8;
+  var foo9 = 9;
+  var foo10 = 10;
   return function () {
+
+    // The number of statements in the inner function does not count toward the
+    // statement maximum.
+
     return 42;
   };
-})()
+}
+```
+
+### ignoreTopLevelFunctions
+
+Examples of **correct** code for this rule with the `{ "max": 10 }, { "ignoreTopLevelFunctions": true }` options:
```

Comment:
Examples of additional **correct** code …

Make sure I am reading this correctly (pardon the pun :) that this option allows same function as in **incorrect** code for the default option?